### PR TITLE
SONARJAVA-2289 Switch instead of if sequence

### PR DESCRIPTION
**Pull request by benzonico**

### CODE REVIEW

File: java-checks/src/main/java/org/sonar/java/checks/SwitchInsteadOfIfSequenceCheck.java [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+      }
+      ifStatementTree = (IfStatementTree) ifStatementTree.elseStatement();
+    }
+    if(level > 2) {
+      reportIssue(tree, "Convert this \"if/else if\" structure into a \"switch\".");
+    }
+
+  }
+
+  private boolean sameEqualCondition(ExpressionTree firstIfCondition, ExpressionTree secondIfCondition) {
```
Comment (SonarTech) [inline java-checks/src/main/java/org/sonar/java/checks/SwitchInsteadOfIfSequenceCheck.java]: MAJOR Make "sameEqualCondition" a "static" method. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: java-checks/src/main/java/org/sonar/java/checks/SwitchInsteadOfIfSequenceCheck.java [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    while (ifStatementTree.elseKeyword() != null && ifStatementTree.elseStatement().is(Tree.Kind.IF_STATEMENT)) {
+      level++;
+      if(!sameEqualCondition(ifStatementTree.condition(), ((IfStatementTree) ifStatementTree.elseStatement()).condition())) {
+        level = 0;
+        break;
+      }
+      ifStatementTree = (IfStatementTree) ifStatementTree.elseStatement();
+    }
+    if(level > 2) {
+      reportIssue(tree, "Convert this \"if/else if\" structure into a \"switch\".");
```
Comment (m-g-sonar) [inline java-checks/src/main/java/org/sonar/java/checks/SwitchInsteadOfIfSequenceCheck.java]: I would add the missing version message as well at the end of the issue message.
```
reportIssue(tree, "Convert this \"if/else if\" structure into a \"switch\"." + context.getJavaVersion().java7CompatibilityMessage());
```

File: java-checks/src/main/java/org/sonar/java/checks/SwitchInsteadOfIfSequenceCheck.java [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    Optional<EqualsOperands> equalsOperandsFirst = getEqualMethodInvocationOperands(firstIfCondition);
+    Optional<EqualsOperands> equalsOperandsSecond = getEqualMethodInvocationOperands(secondIfCondition);
+    return equalsOperandsFirst.isPresent()
+    && equalsOperandsSecond.isPresent()
+    && equalsOperandsFirst.get().identifier.symbol().equals(equalsOperandsSecond.get().identifier.symbol());
+  }
+
+  private static Optional<EqualsOperands> getEqualMethodInvocationOperands(ExpressionTree expressionTree) {
+    if(!expressionTree.is(Tree.Kind.METHOD_INVOCATION)) {
+      return Optional.empty();
```
Comment (benzonico) [inline java-checks/src/main/java/org/sonar/java/checks/SwitchInsteadOfIfSequenceCheck.java]: given : https://jira.sonarsource.com/browse/RSPEC-1698 I doubt we should detect this case...

Comment (benzonico) [inline java-checks/src/main/java/org/sonar/java/checks/SwitchInsteadOfIfSequenceCheck.java]: Ok, handled. Despite the doubt on value...

**Pull request merged**